### PR TITLE
[KHRGA-82] Updated DA Synchronization primitives

### PR DESCRIPTION
**Pull request by PetroBondar**
Issue: KHRGA-82

### CODE REVIEW

File: source/iface/synchronization-primitives.rst [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+  Copyright 2023 The Khronos Group Inc.
   SPDX-License-Identifier: CC-BY-4.0
 
+.. _syncronization-primitives:
+
 **************************
 Synchronization primitives
 **************************
+
+When the user wants to use the :ref:`buffer` simultaneously
```
Comment (mkinsner) [inline source/iface/synchronization-primitives.rst:11]:
```suggestion
When the user wants to use a :ref:`buffer` simultaneously
```

File: source/iface/synchronization-primitives.rst [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+in the SYCL runtime and their own code (e.g. a multi-threaded
+mechanism) and wants to use manual synchronization without
+using a :ref:`host_accessor`, a ``std::mutex`` can be passed
+to the :ref:`buffer` constructor via the right property.
```
Comment (mkinsner) [inline source/iface/synchronization-primitives.rst:15]:
```suggestion
to the :ref:`buffer` constructor through a property.
```

File: source/iface/synchronization-primitives.rst [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
+      std::lock_guard lck { m };
+      // User accesses the data
+      do_something(shD);
+      /* m is unlocked when lck goes out of scope, by normal end of this
+        block but also if an exception is thrown for example */
+    }
+  }
+
+When the runtime releases the ``std::mutex`` the user is guaranteed that the
+data was copied back on the ``std::shared_pointer``, unless the final data
```
Comment (mkinsner) [inline source/iface/synchronization-primitives.rst:36]:
```suggestion
data was copied back to the ``std::shared_pointer``, unless the final data
```

**Comment by mkinsner** [comment]
I'll update the SYCL spec with these wording improvements as well.

**Comment by PetroBondar** [comment]
All comments are fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/102/commits/288a7a213d116ca9321a8c3e5323c08d7ae1dd3a.

**Review by mkinsner — APPROVED** [review]

**Pull request merged**